app/include/bfx-api-cpp/TRequest.hpp

```diff
@@ -0,0 +1,288 @@
+
+      void setAccessKey(string inAccessKey) {
+        accessKey = inAccessKey;
+      }
+
+      void setHeader(map<string, string> inHeader) {
+        header = inHeader;
+      }
+
+    private:
+      ////////////////////////////////////////////////////////////////////////
+      // Private properties
+      ////////////////////////////////////////////////////////////////////////
+      string endpoint = "";
+      string path = "";
+      string secretKey = "";
+      string accessKey = "";
+      string response = "";
+      map<string, string> header;
+      struct curl_slist *curlHeader = nullptr;
+
+      // Curl properties
+      CURL *curlGET;
+      CURL *curlPOST;
+      CURLcode curlStatusCode;
+
+      ////////////////////////////////////////////////////////////////////////
+      // Private methods
+      ////////////////////////////////////////////////////////////////////////
+
+      // Curl write callback function. Appends fetched *content to *userp
+      // pointer. *userp pointer is set up by
+      // curl_easy_setopt(curl, CURLOPT_WRITEDATA, &result) line.
+      // In this case *userp points to result.
+      static size_t writeCallback(
+        void *data,
+        size_t size,
+        size_t nmemb,
+        void *userp) noexcept
+      {
+        (static_cast <string*>(userp))
+          ->append(static_cast <char*>(data), size * nmemb);
+        return size * nmemb;
+      };
+
+      static void getBase64(const string &content, string &encoded) {
+        using CryptoPP::Base64Encoder;
+        using CryptoPP::StringSink;
+        using CryptoPP::StringSource;
+
+        byte buffer[1024] = {};
+
+        for (int i = 0; i < content.length(); ++i)
+            buffer[i] = content[i];
+
+        StringSource ss(
+          buffer,
+          content.length(),
+          true,
+          new Base64Encoder(new StringSink(encoded), false)
+        );
+      };
+
+      static void getHmacSha384(
+        const string &key,
+        const string &content,
+        string &digest)
+      {
+          using CryptoPP::HashFilter;
+          using CryptoPP::HexEncoder;
+          using CryptoPP::HMAC;
+          using CryptoPP::SecByteBlock;
+          using CryptoPP::StringSink;
+          using CryptoPP::StringSource;
+          using CryptoPP::SHA384;
+          using std::transform;
+
+          SecByteBlock byteKey((const byte*)key.data(), key.size());
+          string mac;
+          digest.clear();
+
+          HMAC<SHA384> hmac(byteKey, byteKey.size());
+          StringSource ss1(
+            content, true,
+            new HashFilter(hmac, new StringSink(mac))
+          );
+          StringSource ss2(mac, true, new HexEncoder(new StringSink(digest)));
+          transform(digest.cbegin(), digest.cend(), digest.begin(), ::tolower);
+      };
+
+      void setupHeader() {
+        curlHeader = nullptr;
+        map<string, string>::iterator it;
+        for (it = header.begin(); it != header.end(); it++) {
```

Comment:
Couldn't this code be simplified such ...
delete `map<string, string>::iterator it;`
change `for (auto it = header.begin(); it != header.end(); it++) {`
?
But it's up to you if you prefer more verbose way. `auto` keyword comes handy in these cases.